.github/PULL_REQUEST_TEMPLATE

```diff
@@ -1,10 +1,31 @@
-## What changes were proposed in this pull request?
+<!--  Thanks for sending a pull request!  Here are some tips for you:
+1. If this is your first time, please read our contributor guidelines: https://spark.apache.org/contributing.html
+2. Ensure you have added or ran the appropriate tests for your PR: https://spark.apache.org/developer-tools.html
+3. If the PR is unfinished, add `[WIP]` in your PR title.
+4. Update your PR description whenever it is needed.
+-->
 
-(Please fill in changes proposed in this fix)
+### What changes were proposed in this pull request?
 
-## How was this patch tested?
+### Why changes are needed?
```

Comment:
In the [Kubernates' PR template](https://raw.githubusercontent.com/kubernetes/kubernetes/master/.github/PULL_REQUEST_TEMPLATE.md), the changes and reason descriptions are put in one section:
```
**What this PR does / why we need it**:
```

Just double-check, do you separate the two parts on purpose? I am actually OK with either way.

Reply:
Few small reasons:
- I feel sure some people will ignore one of both questions often :-).
- Wanted to emphasize the reason when a PR adds an API
-  "What changes were proposed in this pull request?" was already existent - wanted to avoid to change it.

But they are rather minor reasons. I don't mind combining it if you and other people more prefer that way.